react/features/recording/functions.js

```diff
@@ -241,3 +256,160 @@ export async function sendMeetingHighlight(state: Object) {
 
     return false;
 }
+
+/**
+ * Starts a local recording.
+ *
+ * @param {Object} store - Redux store.
+ * @returns {void}
+ */
+export async function startLocalRecording(store) {
+    const { dispatch, getState } = store;
+
+    localRecording.recordingData = [];
+    const gdmStream = await navigator.mediaDevices.getDisplayMedia({
+        video: { displaySurface: 'browser' },
+        audio: { channelCount: 2 }
+    });
+    const isBrowser = gdmStream.getVideoTracks()[0].getSettings().displaySurface === 'browser';
+
+    if (!isBrowser) {
+        gdmStream.getTracks().forEach(track => track.stop());
+        throw new Error('You must select the browser tab to record the meeting');
+    }
+
+    initializeAudioMixer();
+    mixAudioStream(gdmStream);
+    const tracks = getTrackState(getState());
+
+    tracks.forEach(track => {
+        if (track.mediaType === MEDIA_TYPE.AUDIO) {
+            const audioTrack = track?.jitsiTrack?.track;
+
+            addAudioTrackToLocalRecording(audioTrack);
+        }
+    });
+
+    localRecording.stream = new MediaStream([
+        ...localRecording.audioDestination.stream.getAudioTracks(),
+        gdmStream.getVideoTracks()[0]
+    ]);
+    localRecording.recorder = new MediaRecorder(localRecording.stream, { mimeType: 'video/webm' });
```

Comment:
Why webm and not mp4? macOS can't even open webm natively.

Reply:
Also, no bitrate configs?

Reply:
Changed it to mp4. What should I add to the bitrate? Isn't the default ok?

Reply:
I don't know what the defaults are, please try to figure out which they are. We might even want to consider config values for them? Not sure.

Reply:
![image](https://user-images.githubusercontent.com/11474153/166465804-985f4704-09c4-4690-8f55-a6b063309dba.png)

From  MDN

Reply:
Sounds good, let's start with that.

Reply:
So, apparently mp4 is not supported on my machine (Chrome on macOS), so I'll leave it as webm. I ran `MediaRecorder.isTypeSupported('video/mp4')` in the browser console and it returns `false`

Reply:
I'd suggest we do something like this: https://github.com/webrtc/samples/blob/a12175269f54c7db9ad10e559055d888a79aab44/src/content/getusermedia/record/js/main.js#L70

Reply:
Ok, I'll try to use mp4 if it's supported, otherwise it'll default to webm